api.go

```diff
@@ -268,7 +284,11 @@ func (api API) Queue(r *http.Request) APIResponse {
 		// 2. Create token for domain
 		token, err := api.Database.PutToken(domain)
 		if err != nil {
-			return APIResponse{StatusCode: http.StatusInternalServerError, Message: err.Error()}
+			return APIResponse{
+				StatusCode:   http.StatusInternalServerError,
+				Message:      err.Error(),
+				TemplatePath: "views/domain.html.tmpl",
+			}
```

Comment:
How do you decide which APIResponses to transform into HTML templates? If this results in an error, what will `template.Execute` return?

Reply:
On `api.go:86`, we only attempt to render HTML if the template is set. Otherwise we respond to HTML requests as usual (with JSON or plain text). Non-200 responses return plain text, so noscript users would see a single line containing `response.Message`.